### PR TITLE
Added classes.md in the kotlin folder

### LINKED ISSUE #1386: [Concept Entry] Kotlin: Classes
### Reason/inspiration (optional)

We would like a new concept entry for [Kotlin](https://www.codecademy.com/resources/docs/kotlin) about classes.

The entry should go in a new file in docs/content/kotlin/concepts/classes/classes.md

The entry should

- Explain what classes are.
- Explain the rules and syntax of declaring a class.
- Have a code block with an example class, showing it in use.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by dakshdeepHERE**
### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

Fixes #1386

### CODE REVIEW

File: content/kotlin/concepts/classes/classes.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        println("Woof!")
+    }
+}
+
+val myDog = Dog("Fido", "Labrador", 3)
+```
+
+This eliminates the need to define default values for the properties and allows you to create objects of the class using the concise `constructor` syntax shown above.
+
+You can also define secondary constructors in a class by using the `constructor` keyword:
```
Comment (SSwiniarski) [inline content/kotlin/concepts/classes/classes.md:95]:
```suggestion
Secondary constructors can also be defined in a class by using the `constructor` keyword:
```

File: content/kotlin/concepts/classes/classes.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    fun bark() {
+        println("Woof!")
+    }
+}
+
+val myDog = Dog("Fido", "Labrador", 3)
+val myPuppy = Dog("Buddy", "Golden Retriever")
+```
+
+This allows you to create objects of the class using different sets of arguments.
```
Comment (SSwiniarski) [inline content/kotlin/concepts/classes/classes.md:124]:
```suggestion
This allows the creation of objects of the class using different sets of arguments.
```

**Comment by dakshdeepHERE** [comment]
@SSwiniarski changes that you requested are done.

**Review by SSwiniarski — APPROVED** [review]
@dakshdeepHERE, looks good for a second review.

File: content/kotlin/concepts/classes/classes.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    var age: Int = 0
+
+    fun bark() {
+        println("Woof!")
+        }
+    }
+}
+```
+
+To create an object of this class, the `new` keyword can be used, or the more concise `constructor` syntax:
```
Comment (KTom101) [inline content/kotlin/concepts/classes/classes.md:58]: I don't think Kotlin has a `new` keyword, can you please re-check this?

File: content/kotlin/concepts/classes/classes.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+```kotlin
+val myDog = Dog()
+val myDog = Dog("Fido", "Labrador", 3)
```
Comment (KTom101) [inline content/kotlin/concepts/classes/classes.md:62]: I believe this line works with the primary constructor you have below. It seems a little confusing having it here.
```suggestion
```

File: content/kotlin/concepts/classes/classes.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+```
+
+Properties and functions of the object can be accessed using the dot notation:
+
+```kotlin
```
Comment (KTom101) [inline content/kotlin/concepts/classes/classes.md:67]: Can we combine this code snippet with the one at line 62 and add the main function? Move line 65 to be with 58.

File: content/kotlin/concepts/classes/classes.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+A primary constructor can also be defined in the class header:
+
+```kotlin
+class Dog(var name: String, var breed: String, var age: Int) {
+    fun bark() {
+        println("Woof!")
+    }
+}
+
```
Comment (KTom101) [inline content/kotlin/concepts/classes/classes.md:82]: Add the main function

File: content/kotlin/concepts/classes/classes.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+        this.name = name
+        this.breed = breed
+        this.age = 0
+    }
+
+    fun bark() {
+        println("Woof!")
+    }
+}
+
```
Comment (KTom101) [inline content/kotlin/concepts/classes/classes.md:112]: Again add main function

**Review by KTom101 — CHANGES_REQUESTED** [review]
@dakshdeepHERE I've added a second review. Please let me know if you have any questions.

File: content/kotlin/concepts/classes/classes.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+}
+```
+
+Here are some rules for defining classes in Kotlin:
+
+- The name of a class should start with a capital letter and should be a noun (e.g. Dog, Person, BankAccount).
+- A class can have properties (data) and functions (behavior).
+- Properties and functions of a class can be made public (accessible from anywhere) or private (accessible only within the class). By default, they are public in Kotlin.
+- A class can have a primary constructor and/or one or more secondary constructors. The primary constructor is defined in the class header, and the secondary constructors are defined using the `constructor` keyword.
+- A class can have an `open`, `abstract`, `sealed`, or `final` modifier. The `open` modifier allows the class to be subclassed, the `abstract` modifier allows the class to have abstract (unimplemented) functions, the `sealed` modifier allows the class to have subclasses, but only in the same file, and the `final` modifier prevents the class from being subclassed.
```
Comment (KTom101) [inline content/kotlin/concepts/classes/classes.md:37]: Is it possible to label where the modifier goes in the syntax? Perhaps label the position in the syntax as `<modifier>`. And mention here that modifiers are optional.

**Comment by KTom101** [comment]
@dakshdeepHERE I've noticed you've added some of the changes already, thank you for that! How's the rest of the changes going?


Also, @SSwiniarski what do you think about the comment I made [here](https://github.com/Codecademy/docs/pull/1421#discussion_r1070503612).

File: content/kotlin/concepts/classes/classes.md (hunk 8 above)
Comment (SSwiniarski) [reply content/kotlin/concepts/classes/classes.md:37]: Agree with mentioning where it goes in the syntax. While I don't see a problem with the modifiers being here, I agree that there should be a Kotlin entry on inheritance.

**Comment by dakshdeepHERE** [comment]
> @dakshdeepHERE I've noticed you've added some of the changes already, thank you for that! Just checking in to see how's the rest of the changes going?
> 
> Also, @SSwiniarski what do you think about the comment I made [here](https://github.com/Codecademy/docs/pull/1421#discussion_r1070503612).

yes yes I will complete everything by tomorrow I'm just having very busy week in college :d

File: content/kotlin/concepts/classes/classes.md (hunk 8 above)
Comment (KTom101) [reply content/kotlin/concepts/classes/classes.md:37]: Ok, we'll leave the modifiers here and just add where the modifiers would be written in the syntax. 

Also, I'll open up an issue for a Kotlin entry on inheritance. 

cc @dakshdeepHERE

**Comment by KTom101** [comment]
> > @dakshdeepHERE I've noticed you've added some of the changes already, thank you for that! Just checking in to see how's the rest of the changes going?
> 
> > 
> 
> > Also, @SSwiniarski what do you think about the comment I made [here](https://github.com/Codecademy/docs/pull/1421#discussion_r1070503612).
> 
> 
> 
> yes yes I will complete everything by tomorrow I'm just having very busy week in college :d

@dakshdeepHERE No worries, was just checking in. Thank you for responding so quickly!

File: content/kotlin/concepts/classes/classes.md (hunk 3 above)
Comment (dakshdeepHERE) [reply content/kotlin/concepts/classes/classes.md:58]: yes correct its there in java. So I will remove the `new` keyword and continue with the `constructor`

**Comment by KTom101** [comment]
@dakshdeepHERE Just checking in again to see if there is anything I can help with.

**Review by KTom101 — APPROVED** [review]
@dakshdeepHERE Due to no response, I've added the changes and fixed the failed check [cbfbb8f](https://github.com/Codecademy/docs/pull/1421/commits/cbfbb8f1e5c0a492b4ab95caa85b3813f7f73091).

@SSwiniarski Any other changes before we merge?

**Review by SSwiniarski — APPROVED** [review]
@KTom101, @dakshdeepHERE Looks good to merge.

**Comment by KTom101** [comment]
@dakshdeepHERE Thanks for contributing, entry is live https://www.codecademy.com/resources/docs/kotlin/classes 🎉 You're all set!
Screen Shot 2023-02-18 at 1 35 55 PM

**Pull request merged**